### ISSUE TITLE
Bug: Balances persists when transactions are deleted

### ISSUE THREAD

**Issue by ptsimon** [oss-apps/split-pro#130]
I have an issue here with my group. I have added some transactions in a group before that led me to a balance, then deleted the all the group  transactions hoping for the balance to be zero. Nope it did not despite having empty transactions in the group. I can't delete the group as well because i have to "settle balances" first before i can delete.

![Screenshot_2024-10-24-13-09-27-798_com android chrome](https://github.com/user-attachments/assets/4d04f03a-5aac-4e04-bdb0-f143b82daa3f)

**Comment by KMKoushik** [comment]
yes this is an issue with group expense calculation but it won't actually affect the balance between your friends, if you go to balances page it should be calculated correctly.

this is a problem with group where i don't know which balance get canceled and i recalculate those only on settle up events for hroups. 

may be i can handle this for delete event. that being said you don't have to worry about this. once you settle up a payment this will be reset for the group!

**Comment by alexanderwassbjer** [comment]
I cant really reproduce this bug? Even how many times I have tried to add and remove expenses with multiple people in a group I cant seem to reproduce the issue?

**Comment by ptsimon** [comment]
Hmm, if i remember correctly, i added expenses in the group, then i deleted
the transactions in the Activity tab

**Comment by alexanderwassbjer** [comment]
@ptsimon Thanks

@KMKoushik I saw that when clicking in the activity tab on an expense its always redirecting to /expenses/id.
If it would redirect to /groups/id/expenses/id in group expense type and /expenses/id on friend expense type, and the delete would be from that page group expense page instead I think it would work with the calculation, or? 😃
